### PR TITLE
Fix empty OSC 2 sequence not clearing terminal title

### LINKED ISSUE #312403: Empty OSC 2 escape sequence does not clear ${sequence} terminal title variable
## Repro

1. Set `"terminal.integrated.tabs.title": "${sequence}${separator}${process}"` in settings
2. In the integrated terminal, emit an OSC 2 title: `printf '\033]2;my-title\007'`
3. Tab now shows "my-title - zsh" — `${sequence}` picks it up correctly
4. Emit an empty OSC 2 to clear: `printf '\033]2;\007'`
5. Tab still shows "my-title - zsh" — `${sequence}` retains the stale value

## Expected

Step 4 should clear `${sequence}` back to empty, making the tab revert to just "zsh" (from `${process}`). Per the [xterm spec](https://invisible-island.net/xterm/ctlseqs/ctlseqs.html), OSC 2 with an empty string body is a valid sequence that sets the window title to the empty string.

## Actual

`${sequence}` keeps its last non-empty value indefinitely. There's no way to clear it via escape sequences once set.

## Root cause

xterm.js handles this correctly — it fires `onTitleChange("")` for empty OSC 2. The problem is in VS Code's handler. `_updateTitleProperties` in `terminalInstance.ts` starts with:

```typescript
if (!title) {
    return this._processName;
}
```

Empty string is falsy in JavaScript, so this early-returns without ever reaching `this._sequence = title`. The stored sequence is never cleared.

## Prior art

Windows Terminal had the identical bug where an empty OSC 2 didn't clear the title in [microsoft/terminal#16784](https://github.com/microsoft/terminal/issues/16784), and it was resolved there.

## Suggested fix

Change the guard from `if (!title)` to `if (title === undefined)` (or `if (title == null)`). This lets empty strings through to actually clear `_sequence` while still handling the undefined/null case.

## Workaround

Emit a non-empty string (e.g., the shell name) instead of an empty OSC 2 to overwrite the stale value.

### PR DESCRIPTION
**Pull request by wpfleger96**
When a shell sends an empty OSC 2 sequence (`\033]2;\007`), the `${sequence}` terminal title variable should reset to `""` and the tab should revert to the process name. Instead, `_sequence` retained its last non-empty value permanently.

The guard in `_updateTitleProperties` used `if (!title)`, which treats empty string as falsy and returns before the `TitleEventSource.Sequence` case can assign `this._sequence = ""`. xterm.js correctly fires `onTitleChange("")` for empty OSC 2 — the bug was entirely in VS Code's handler. Changed to `if (title == null)` so empty strings flow through to the switch while still guarding `undefined`/`null`.

Since the broader guard also lets empty strings through the Api path, `rename("")` (reachable from the rename input box) would now set `_staticTitle = ""` and `_titleSource = Api`, blocking future process-name updates. Added a normalization in `rename()` that converts empty/whitespace-only titles to `undefined`, so the existing null guard handles it as a reset — matching `validateTerminalName`'s stated intent.

- `terminalInstance.ts`: change guard from `!title` to `title == null`
- `terminalInstance.ts`: normalize empty `rename()` input to `undefined`
- `terminalInstance.test.ts`: add test for empty sequence falling back to process name

Fixes #312403

### CODE REVIEW

**Comment by wpfleger96** [comment]
@microsoft-github-policy-service agree

File: src/vs/workbench/contrib/terminal/test/browser/terminalInstance.test.ts [hunk 1]
```diff
@@ -475,6 +475,12 @@ suite('Workbench - TerminalInstance', () => {
 			strictEqual(terminalLabelComputer.title, 'sequence');
 			strictEqual(terminalLabelComputer.description, 'sequence');
 		});
+		test('should resolve empty sequence to process name', () => {
+			const terminalLabelComputer = createLabelComputer({ terminal: { integrated: { tabs: { separator: ' - ', title: '${sequence}${separator}${process}', description: '${sequence}' } } } });
+			terminalLabelComputer.refreshLabel(createInstance({ capabilities, processName: 'zsh', sequence: '' }));
```
[the remark this answers is not in the document]
Comment (wpfleger96) [reply src/vs/workbench/contrib/terminal/test/browser/terminalInstance.test.ts:480]: yeah this is fair  - the `TerminalLabelComputer` tests use mock instances with pre-set properties, so this would've passed before the guard change too. I looked into writing a test that goes through `_updateTitleProperties` directly, but that requires a full `TerminalInstance` with xterm/process manager wired up which is pretty heavy for a one-line guard fix. I kept this test because it still documents the expected label behavior when sequence is empty (template fallback + separator suppression), but I agree it's not a true regression test for the bug.

File: src/vs/workbench/contrib/terminal/browser/terminalInstance.ts [hunk 2]
```diff
@@ -2101,7 +2101,7 @@ export class TerminalInstance extends Disposable implements ITerminalInstance {
 	}
 
 	private _updateTitleProperties(title: string | undefined, eventSource: TitleEventSource): string {
-		if (!title) {
+		if (title == null) {
 			return this._processName;
 		}
```
[the remark this answers is not in the document]
Comment (wpfleger96) [reply src/vs/workbench/contrib/terminal/browser/terminalInstance.ts:2113]: good catch - I traced through this and `rename("")` is reachable from the rename input box (`validateTerminalName` shows an info message but doesn't block submission), and with the `title == null` guard it now flows into the `Api` case, `sets _staticTitle = ""` / `_titleSource = Api`, and blocks future process-name updates. 

Added a normalization in `rename()` that converts empty input to undefined so it hits the null guard and resets as intended - which is what `validateTerminalName`'s info message already tells the user will happen. See 8ed794e.

**Review by meganrogge — DISMISSED** [review]

**Comment by wpfleger96** [comment]
👋 @meganrogge I just rebased the PR and addressed the `ESLint` CI failure if you don't mind reviewing again!

**Review by benvillalobos — APPROVED** [review]

**Review by anthonykim1 — APPROVED** [review]

**Comment by wpfleger96** [comment]
@meganrogge @benvillalobos @anthonykim1 thank you!! 

😅  I'll need another approval for the CI workflows to rerun too before this can merge whenever you have a minute!

**Review by benvillalobos — APPROVED** [review]

**Comment by wpfleger96** [comment]
hmmm I'm not sure why the [Monaco Editor checks](https://github.com/microsoft/vscode/actions/runs/25761647933/job/75830541431) workflow was cancelled? 

@benvillalobos are you able to approve the workflows one more time and enable auto merge for the PR for me? it looks like I don't have permission to and I'd love to land this once the checks pass again without having to ping anyone for yet another review 😅
